CSharpMinifier/IdentifierGenerator.cs

```diff
@@ -0,0 +1,102 @@
+            "is", "lock", "long", "namespace", "new", "null", "object", "operator", "out",
+            "override", "params", "private", "protected", "public", "readonly", "ref", "return", "sbyte",
+            "sealed", "short", "sizeof", "stackalloc", "static", "string", "struct", "switch",
+            "this", "throw", "true", "try", "typeof", "uint", "ulong", "unchecked", "unsafe",
+            "ushort", "using", "virtual", "void", "volatile", "while"
+        };
+
+        public IdentifierGenerator()
+        {
+            RenamedVariables = new Dictionary<VariableDeclaratorSyntax, string>();
+            RenamedMethods = new Dictionary<MethodDeclarationSyntax, string>();
+            RenamedTypes = new Dictionary<ClassDeclarationSyntax, string>();
+            RenamedProperties = new Dictionary<PropertyDeclarationSyntax, string>();
+        }
+
+        public string GetNextName(SyntaxNode node)
+        {
+            if (_existingNames.Count > 0)
+            {
+                string lastName = _existingNames[_existingNames.Count - 1];
+                string nextName = IncrementName(lastName);
+                _existingNames.Add(nextName);
+                AddToDictionary(node);
+                return nextName;
+            }
+            var firstSymbol = Char.ConvertFromUtf32(FirstSymbolCode);
+            _existingNames.Add(firstSymbol);
+            AddToDictionary(node);
+            return firstSymbol;
+        }
+
+        private void AddToDictionary(SyntaxNode node)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.MethodDeclaration:
+                    RenamedMethods.Add((MethodDeclarationSyntax)node, LastName);
+                    break;
+                case SyntaxKind.VariableDeclarator:
+                    RenamedVariables.Add((VariableDeclaratorSyntax)node, LastName);
+                    break;
+                case SyntaxKind.ClassDeclaration:
+                    RenamedTypes.Add((ClassDeclarationSyntax)node, LastName);
+                    break;
+                case SyntaxKind.PropertyDeclaration:
+                    RenamedProperties.Add((PropertyDeclarationSyntax)node, LastName);
+                    break;
+            }
+        }
+
+
+        private string LastName => _existingNames.LastOrDefault();
+
+        private string IncrementName(string lastName)
+        {
+            if (lastName.Length <= 0)
+                lastName = "a";
+            char lastChar = lastName[lastName.Length - 1];
+            string fragment = lastName.Substring(0, lastName.Length - 1);
+            while (lastChar < LastSymbolCode)
+            {
+                lastChar++;
+                if (_cSharpKeywords.Contains(fragment + lastChar))
+                    continue;
+                return fragment + lastChar;
+            }
+            return IncrementName(fragment) + Char.ConvertFromUtf32(FirstSymbolCode);
```

Comment:
Try to rewrite this method in a more clear way:

* Get rid of recursion.
* Remove `Char.ConvertFromUtf32`. It's possible to use `ToString` method here.